src/Microsoft.Dnx.Watcher.Core/ProcessWatcher.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Dnx.Watcher.Core
+{
+    public class ProcessWatcher
+    {
+        private Process _runningProcess;
+
+        public int Start(string executable, string arguments, string workingDir)
+        {
+            // This is not thread safe but it will not run in a multithreaded environment so don't worry
+            if (_runningProcess != null)
+            {
+                throw new InvalidOperationException("The previous process is still running");
+            }
+
+            // TODO: check that it is running 
+            _runningProcess = new Process();
+            _runningProcess.StartInfo = new ProcessStartInfo()
+            {
+                FileName = executable,
+                Arguments = arguments,
+                UseShellExecute = false,
+                WorkingDirectory = workingDir
+            };
+
+            WorkaroundDnxInternalEnvironmentVariables(_runningProcess.StartInfo);
+
+            _runningProcess.Start();
+
+            return _runningProcess.Id;
+        }
+
+        public async Task<int> WaitForExitAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                await Task.Run(() =>
+                {
+                    while (true)
+                    {
+                        if (_runningProcess.WaitForExit(500))
+                        {
+                            break;
+                        }
+
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            if (!_runningProcess.HasExited)
+                            {
+                                _runningProcess.Kill();
+                            }
+
+                            break;
+                        }
+                    }
+                });
+
+                return _runningProcess.ExitCode;
+            }
+            finally
+            {
+                _runningProcess = null;
+            }
+        }
+
+        // This must be removed while DNX doesn't pass environment variables anymore
+        private static void WorkaroundDnxInternalEnvironmentVariables(ProcessStartInfo procStartInfo)
+        {
+            string[] _environmentVariablesToRemove = new string[]
+            {
+                "DNX_DEFAULT_LIB",
+                "DNX_APPBASE",
+                "DNX_COMPILATION_SERVER_PORT",
```

Comment:
I have to remove that environment variable. Otherwise both DNXs will use the same port and the child (the process being watched) will fail:

```
System.IO.FileLoadException: Could not load file or assembly 'ConsoleApp1' or one of its dependencies. General Exception (Exception from HRESULT: 0x80131500)
File name: 'ConsoleApp1' ---> System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: The context of the project at c:\users\victorhu\documents\visual studio 2015\Projects\watch\ConsoleApp1\src\ConsoleApp1 could not be determined. This can happen if a project references other projects by source, and those projects have a global.json file specifying a different version of the SDK.
   at Microsoft.Dnx.Compilation.DesignTime.DesignTimeHostCompiler.<Compile>d__7.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.Dnx.Compilation.DesignTime.DesignTimeHostProjectCompiler.CompileProject(CompilationProjectContext projectContext, Func`1 referenceResolver, Func`1 resourcesResolver)
   at Microsoft.Dnx.Compilation.LibraryExporter.<>c__DisplayClass18_0.<ExportProject>b__0(CacheContext ctx)
   at Microsoft.Dnx.Compilation.Caching.CacheExtensions.<>c__DisplayClass0_0`1.<Get>b__0(CacheContext ctx)
   at Microsoft.Dnx.Compilation.Caching.Cache.CreateEntry(Object k, Func`2 acquire)
   at Microsoft.Dnx.Compilation.Caching.Cache.<>c__DisplayClass5_0.<AddEntry>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Dnx.Compilation.Caching.Cache.Get(Object key, Func`2 factory)
   at Microsoft.Dnx.Compilation.Caching.CacheExtensions.Get[T](ICache cache, Object key, Func`2 factory)
   at Microsoft.Dnx.Compilation.LibraryExporter.ExportProject(ProjectDescription project, String aspect, LibraryExporter exporter)
```